### DISCUSSION TITLE
Check solving state of model

### DISCUSSION THREAD

**Discussion by Zhanwei-Liu** [metab0t/PyOptInterface#5] [Q&A] [answered]
I am wondering if it is possible to automatically check the solving status of the model, similar to the examples from `GurobiPy`：
```python

from gurobipy import GRB

if model.Status == GRB.INF_OR_UNBD:
    pass
if model.Status == GRB.OPTIMAL:
    pass
elif model.Status != GRB.INFEASIBLE:
    pass
```
I know I can get the solving state via 
```python
sol_state = model.get_model_attribute(poi.ModelAttribute.TerminationStatus)
```
, but I am unsure what should be compared with `sol_state`, a `string` or `integer`？ Could you provide a small example? Thank you!

**Comment by metab0t** [comment, marked as answer]
For the solver-agnostic attribute, the return value of `model.get_model_attribute(poi.ModelAttribute.TerminationStatus)` is [`pyoptinterface.TerminationStatusCode`](https://metab0t.github.io/PyOptInterface/api/pyoptinterface.html#pyoptinterface.TerminationStatusCode) as documented in the [table](https://metab0t.github.io/PyOptInterface/model.html#id1).
```Python
status = model.get_model_attribute(poi.ModelAttribute.TerminationStatus)
if status == poi.TerminationStatusCode.OPTIMAL:
    ...
elif status == poi.TerminationStatusCode.INFEASIBLE:
    ...
elif status == poi.TerminationStatusCode.INFEASIBLE_OR_UNBOUNDED
    ...
```

The concept is the same as JuMP.jl and I have followed the same logic in Gurobi.jl
https://jump.dev/JuMP.jl/stable/manual/solutions/#Why-did-the-solver-stop?
https://jump.dev/JuMP.jl/stable/moi/reference/models/#MathOptInterface.TerminationStatusCode

Of course, you can also query the Gurobi-specific status as documented in https://metab0t.github.io/PyOptInterface/gurobi.html#attribute and https://www.gurobi.com/documentation/current/refman/attributes.html#sec:Attributes
```Python
import pyoptinterface as poi
from pyoptinterface import gurobi
GRB = gurobi.GRB

status = model.get_model_raw_attribute("Status")
if status == GRB.INF_OR_UNBD:
    pass
if status == GRB.OPTIMAL:
    pass
elif status == GRB.INFEASIBLE:
    pass
```

The [`GRB`](https://github.com/metab0t/PyOptInterface/blob/master/lib/gurobi_model_ext_constants.cpp) submodule has exported most constants like [`GRB`](https://www.gurobi.com/documentation/current/refman/py_constants.html) in gurobipy and I will figure out how to introduce it in our documentation.

**Comment by metab0t** [comment]
@Zhanwei-Liu I have updated the [code](https://github.com/metab0t/PyOptInterface/commit/ef924044ee68045218aa550d07dabb8507004956) and [documentation](https://github.com/metab0t/PyOptInterface/commit/a607d1db0b600d0f29ef6193f69afded1711a8c0) for `pyoptinterface.gurobi.GRB` and `pyoptinterface.copt.COPT`. Now they are exactly identical to `gurobipy.GRB` and `coptpy.COPT` because I write a script to ensure that.

**Reply by Zhanwei-Liu** [reply]
Great! Thank you for your update!
